### DISCUSSION TITLE
Post PGN traversal (prev and next) function call

### DISCUSSION THREAD

**Discussion by coffeeDev98** [mliebelt/pgn-viewer#381] [Q&A]
Hi @mliebelt, 
Is there any way to expose a method to call a function that get's executed after the PREV and NEXT buttons are pressed. I've managed to extract move object after a move is played by using the PGNBASE function directly from the library instead of PGNEDIT and recreate PGNEDIT function within the file I want to use it in, outside the library. as given below: 

```
  const lastStats: any = {};
  let lastStatsPgn = "";
  let lastStatsLastMove = "";

  Object.defineProperty(lastStats, "pgn", {
    get() {
      return lastStatsPgn;
    },
    set(val) {
      lastStatsPgn = val;                         //extracts pgn value from library
    },
  });
  Object.defineProperty(lastStats, "lastMove", {
    get() {
      return lastStatsLastMove;
    },
    set(val) {
      console.log("LAST_MOVE: ", val);
      lastStatsLastMove = val;                 // extracts last move object from library
    },
  });

  const pgnEdit = function (boardId: any, configuration: any) {
    let base = pgnBase(
      boardId,
      Object.assign(
        {
          showFen: true,
          mode: "edit",
          movable: {
            free: false,
            events: {
              after: function (orig: any, dest: any, meta: any) {
                base.onSnapEnd(orig, dest, meta);
                lastStats.pgn = base.getPgn().writePgn();  //custom function called after a move is played
                lastStats.lastMove = base.getPgn().findMove(board.getFen()); //custom function called  a move is played
              },
            },
          },
          viewOnly: false,
        },
        configuration
      )
    );
    base.generateHTML();
    base.generateMoves();
    let board = base.generateBoard();
    return { base, board };
  };
```

The goal is to call a custom function to get the move object post a PREV or NEXT button click

![image](https://user-images.githubusercontent.com/63295908/175859934-9771b882-7c14-4dd7-95c6-a21f6e8fcf77.png)

Move Object:
```
{
    "from": "g7",
    "to": "g5",
    "notation": {
        "notation": "g5",
        "col": "g",
        "row": "5",
        "ep": false
    },
    "variations": [],
    "turn": "b",
    "moveNumber": 1,
    "fen": "rnbqkbnr/pppppp1p/8/6p1/7P/8/PPPPPPP1/RNBQKBNR w KQkq g6 0 2",
    "prev": 0,
    "index": 1
}
```

**Comment by mliebelt** [comment]
Got it, and sorry, I am busy at the moment with other stuff, no chance to have anything started the next 2 weeks. For me, that correlates heavily to the 2 things I want to add next:

* Clear, understandable, documented API for viewer, reader, ...
* Notification for the relevant "actions" that are possible with the viewer.

**Reply by coffeeDev98** [reply]
No problem, I understand. 
This might be a hacky way... but is it possible to pass a third parameter to pgnBase to manage notifications for relevant actions?... as a temporary implementation. If this is a viable approach, then I can implement this in my project for now...

**Reply by mliebelt** [reply]
I would like to get something (rudimentary, but from design ok) up and running. The goal is, that currently available hooks/notifications from Chessground are supported, and in a similar fashion, additional hooks/notifications are done for the relevant actions done in the viewer. I just have to collect them, do a rudimentary design, and implement the first few. This should be part of the configuration, so that it is much easier to integrate it, without going into an own implementation on base of `pgnBase` (that should be an implementation detail, and not an API).
